scilifelab/lims_utils/workset_upload.py

```diff
@@ -63,42 +76,80 @@ def __init__(self, lims, crawler):
         outs=crawler.starting_proc.all_outputs()
         for out in outs:
             if out.type == "Analyte" and len(out.samples) == 1 :
-                self.name.add(out.location[0].name)
-        self.obj['name']=self.name.pop()
+                try:
+                    self.name.add(out.location[0].name)
+                except:
+                    self.log.warn("no name found for workset {}".format(out.id))
+
+        try:      
+            self.obj['name']=self.name.pop()
+        except:
+            self.log.error("No name found for current workset {}, might be an ongoing step.".format(crawler.starting_proc.id))
+            raise NameError
+        self.obj['technician']=crawler.starting_proc.technician.initials
         pjs={}
         for p in crawler.projects:
             pjs[p.id]={}
             pjs[p.id]['name']=p.name
-            if "library" in p.udf:
+            try:
                 pjs[p.id]['library']=p.udf['Library construction method']
-            if "application" in p.udf:
+            except KeyError:
+                pjs[p.id]['library']=None
+            try:
                 pjs[p.id]['application']=p.udf['Application']
+            except KeyError:
+                pjs[p.id]['application']=None
+
             pjs[p.id]['samples']={}
             for sample in crawler.samples:
                 if sample.project == p:
                     pjs[p.id]['samples'][sample.name]={}
                     pjs[p.id]['samples'][sample.name]['library']={}
                     pjs[p.id]['samples'][sample.name]['sequencing']={}
-                    for lib in crawler.libaggre:
+                    try:
+                        pjs[p.id]['samples'][sample.name]['customer_name']=sample.udf['Customer Name']
+                    except KeyError:
+                        pjs[p.id]['samples'][sample.name]['customer_name']= None
+
+
+                    pjs[p.id]['samples'][sample.name]['rec_ctrl']= {}
+                    for i in crawler.starting_proc.all_inputs():
+                        if sample in i.samples:
+                            pjs[p.id]['samples'][sample.name]['rec_ctrl']['status']=i.qc_flag
```

Comment:
no, this looks for the input of the setup workset plate associated with the sample we are currently looking into and creating a reception control status key associated with the current sample. This is not a new line, it's just a long one.